app/selectors/networkController.ts

```diff
@@ -197,6 +205,52 @@ export const selectNetworkConfigurations = createDeepEqualSelector(
   },
 );
 
+/**
+ * Gets EVM (and eventually non-EVM) Network Configurations keyed by CaipChainId.
+ *
+ * @returns network configurations keyed by CaipChainId.
+ */
+// Uncomment relevant lines when ready for non-evm support
+export const getNetworkConfigurationsByCaipChainId = (
+  evmNetworkConfigurationsByChainId: Record<Hex, NetworkConfiguration>,
+  // nonEvmNetworkConfigurationsByChainId,
+): Record<CaipChainId, EvmAndMultichainNetworkConfigurationsWithCaipChainId> => {
+    const networkConfigurationsByCaipChainId: Record<CaipChainId, EvmAndMultichainNetworkConfigurationsWithCaipChainId> = {
+    };
+
+    Object.entries(evmNetworkConfigurationsByChainId).forEach(([chainId, networkConfiguration]) => {
+      const caipChainId: CaipChainId = `eip155:${parseInt(chainId, 16)}`;
+      networkConfigurationsByCaipChainId[caipChainId] = {
+        ...networkConfiguration,
+        caipChainId
+      };
+    });
+
+    // Object.entries(nonEvmNetworkConfigurationsByChainId).forEach(([_caipChainId, networkConfiguration]) => {
```

Comment:
Maybe lets add a comment explaining why we're leaving this commented out code here
```suggestion
    // for use in the near future when we want to include nonEvm configurations in what's returned
    // Object.entries(nonEvmNetworkConfigurationsByChainId).forEach(([_caipChainId, networkConfiguration]) => {
```

Reply:
there's a comment at the top of this function that mentions this already, but i'll add this anyway